projects/Mallard/ophan/src/commonMain/kotlin/ophan/Ophan.kt

```diff
@@ -40,27 +42,61 @@ class OphanApi(
             userId,
             logger,
             FileRecordStore(recordStorePath)
-    ))
+    ), logger)
 
-    fun sendTestAppScreenEvent(screenName: String, eventId: String) {
+    fun componentEventBuilder(componentType: String, action: String, eventId: String, value: String?, componentId: String?): Event {
         val event = Event.Builder()
                 .eventId(eventId)
                 .eventType(EventType.COMPONENT_EVENT)
                 .viewId(null) /* TODO */
                 .componentEvent(ComponentEvent.Builder()
                         .component(ComponentV2.Builder()
-                                .componentType(ComponentType.APP_SCREEN)
-                                .id(screenName)
+                                .componentType(ComponentType.valueOf(componentType))
+                                .id(componentId) 
                                 .products(emptySet())
                                 .campaignCode(null)
                                 .labels(emptySet())
                                 .build()
                         )
-                        .action(Action.VIEW)
-                        .value("yes")
+                        .action(Action.valueOf(action))
+                        .value(value)
                         .build()
                 )
                 .build()
+        return event;
+    }
+
+    fun sendAppScreenEvent(screenName: String, value: String?, eventId: String) {
+        val event = this.componentEventBuilder("APP_SCREEN", "VIEW", eventId, value, screenName)
```

Comment:
Constants such as "APP_SCREEN" and "VIEW" should really be declared as such on the class so that we're not creating new string objects every time the function is invoked.

Reply:
Cool - I didnt know about this optimisation. I will add this to the Native tidy up ticket.